src/github.com/matrix-org/dendrite/clientapi/routing/device.go

```diff
@@ -35,6 +36,10 @@ type devicesJSON struct {
 	Devices []deviceJSON `json:"devices"`
 }
 
+type deviceUpdateJSON struct {
+	DisplayName string `json:"display_name"`
```

Comment:
I'd probably make this a `*string` so that nulls get inserted into the DB rather than empty strings when a device display name isn't specified